src/wallet/coinselection.cpp

```diff
@@ -299,7 +299,7 @@ bool KnapsackSolver(const CAmount& nTargetValue, std::vector<OutputGroup>& group
 void OutputGroup::Insert(const CInputCoin& output, int depth, bool from_me, size_t ancestors, size_t descendants) {
     m_outputs.push_back(output);
     m_from_me &= from_me;
-    m_value += output.effective_value;
```

Comment:
this only isn't a reachable "bug" because before this PR the values are the same, yes?

Reply:
hmm actually no, this happens for any `OutputGroup`, which includes Bnb?

Reply:
it isn't reachable because the values are the same as the outputs' effective values aren't ever actually to their effective values.

Reply:
ah I see, it's being set externally in `SelectCoinsMinConf`. Ok, this set of changes makes it much easier to understand.